kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -49,15 +52,38 @@ def _hash(string: str) -> str:
     return hashlib.sha512(bytes(string, encoding="utf8")).hexdigest()
 
 
-def _get_hashed_username():
+def _get_or_create_uuid():
+    """
+    Reads a UUID from a configuration file or generates and saves a new one if not present.
+    """
+    config_path = user_config_dir("kedro")
+    full_path = os.path.join(config_path, CONFIG_FILENAME)
+    config = ConfigParser()
```

Comment:
About the use of `ConfigParser`, more opinions needed https://github.com/kedro-org/kedro-plugins/issues/333#issuecomment-1981613714

Reply:
We decided to go for TOML